### PR TITLE
Extract self-extracting archives for searchability

### PR DESCRIPTION
**Pull request by majick**
Where possible, archive contents were left in the same directory where the archive itself existed.

Several archives contained files with ambiguous names such as "README" and editorially it made more sense to make a subdirectory to extract those into.  This is probably the most controversial part of the commit as it is a small structural change.

EMSIG.400 in the 03-TELCOM was moved.  The self-extracting archive contained a byte identical file and I elected instead to keep that copy in EMSIG/ instead of the existing file in the flat directory.

### CODE REVIEW

**Comment by majick** [comment]
Several of the executables in the archive are self-extracting archives containing other files, some of which are searchable text that describes or documents the contents of the archive file itself.  Some of those contents are untokenized BASIC source or assembly.  As in #1 these should probably be searchable and browsable without having to extract binary archives.

A modest but significant editorial choice was made here, so again I'm electing not to merge cold: in cases where the archive contained a file with an ambiguous "README" or "READ.ME" file name that would otherwise be hard to associate with the archive if just dumped into the pile of other files, I used a subdirectory.

**Comment by bkw777** [comment]
This does require some thought. Probably the original single-depth structure including archive files was because these were download collections on Compuserve, and there was (probably) no such thing as a folder there.

Probably today we can just forget about that original context and allow subdirectories.

Spome questions to consider I see are:
- Adopt a consistent rule and always do the same thing for any archive?: Meaning, even if the contents would not collide with something in the main directory, still always create a folder and place all contents in it. Or handle them case by case. Like what would be the point for single-file archives that don't conflict?
- Keep the original archives even if there is a directory by the same name? Reason against, obviously, not needed. Reason for, maybe some text file referes to the archive file by name, maybe still no problem as long as the new directory is the same as the archive name, including extension. But then you cannot retain the original archive if the directory has the same name. You must delete it or at least move it somewhere else (which would just be messy).
- reduce ambiguity and lost info but increase ugliness by naming the directories foo_ or foo_unpacked or something, so that both the original archive and it's contents may exist together in the same directory.
- adding/removing files from the main directories will mean that the .CAT files will grow out of sync. Do we edit them? I think I would prefer not, but I think it could be argued either way. I did always intend for the contents to change over time, thus "living" right in the name to remove any possible doubt. But so far almost nothing has changed so it hasn't come up, so I haven't developed a firm and conscious stance.
- Basically anything about the original files  that would constitute any lost information if an archive were replaced by it's unpacked contents. I don't know what that might be but I want to try hard to think of any for more than 2 minutes before deciding there are none.

**Comment by majick** [comment]
I thought this would be worthy of some consideration, too, since it introduces structural change.  A handful of notes that should be thought about before anyone pushes the green button on this:

- No original, as-archived files were harmed in this commit.  One file was moved, because it duplicated the file in the archive from whence it came, but that's it.  Larger policy on that matter is not considered or proposed.
- In all cases, where a subdirectory was created, it was named as the archive it was extracted from, without the "extension"
- The unpacked archives do omit, by nature, at least one type of data: archive file metadata.  In the case of the ZIP files from the previous PR as well as some of the SFX files here, that's mostly the "comments" part of the archive.
- This omitted data is a very strong case to continue to keep both original and extracted files in-repo
- In MMDEMO.EXE, that comment metadata actually extracts as a file called `!` because that's how LHarc files work.  I renamed it to DESC.TXT since that's a bit easier for most tools and systems to work with and is more recognizable and descriptive to humans.

On the larger matter of policy, maybe that should be a bug or you could just email me or something.  Whatever works.

**Comment by bkw777** [comment]
If the archive names without extension didn't happen to conflict, and you've got them all, then I guess that makes that question easy.  The original archive and the unpacked directory may coexist that way. Any text files referencing the .exe or whatever are still ok because the original file is still there.

It also means the filename for extracted metadata isn't too critical since the original archive didn't go anywhere.

**Comment by majick** [comment]
That was the philosophy that more-or-less drove how I approached this: preserve what's there while adding value and making it clear what the new stuff is.  Make the new stuff as clearly and intuitively as possible associated to the existing stuff.  Stick to existing convention as much as possible.

**Comment by majick** [comment]
Additional commit addresses #4 to include ARC files, which I totally forgot were a thing from back in the day.

**Comment by bkw777** [comment]
looks good
